base.js

```diff
@@ -65,7 +87,46 @@ class Base extends EventEmitter {
       confPath = envConfPath
     }
 
-    _.merge(this.conf, this.getConf(this.ctx.env, group, confPath))
+    const exampleConfigPath = `${confPath}.example`
+    const config = this.getConf(this.ctx.env, group, confPath)
+
+    if (fs.existsSync(exampleConfigPath)) {
+      const exampleConfig = this.getConf(this.ctx.env, group, exampleConfigPath)
+      const groupedExampleCfg = _.get(exampleConfig, group, exampleConfig)
+      const srcCfgKeys = extractObjectKeys(groupedExampleCfg)
+
+      const groupedCfg = _.get(config, group, config)
+      const destCfgKeys = extractObjectKeys(groupedCfg)
+
+      const missingKeys = _.difference(srcCfgKeys, destCfgKeys)
+
+      if (missingKeys.length) {
+        printOutput('CONFIG MISSING KEY/VALUE FROM CONFIG.EXAMPLE', `[${missingKeys}] missing in ${confPath}`)
+        process.exit(1)
+      }
+
+      if (validation) {
+        for (let key in validation) {
+          const check = validation[key]
+          if (check.required && !_.get(groupedCfg, key)) {
+            printOutput('CONFIG MISSING MANDATORY VALUE', `['${key}'] missing value`)
+            process.exit(1)
+          }
+
+          if (check.sameAsExample) {
+            const cfgValue = _.get(groupedCfg, key)
+            const exampleCfgValue = _.get(groupedExampleCfg, key)
+
+            if (!_.isEqual(cfgValue, exampleCfgValue)) {
+              printOutput('CONFIG VALUE MISMATCH', `config['${key}']:[${cfgValue}] !== exampleConfig['${key}']:[${exampleCfgValue}]`)
+              process.exit(1)
+            }
+          }
+        }
+      }
+    }
+
+    _.merge(this.conf, config)
```

Comment:
I understand this replaces all configuration by values from the example config.
Is this intended?

Reply:
this was not touched. I assume it works as expected as a lot of services would fail otherwise